### PR TITLE
18 inheritancefree first implementation

### PR DESCRIPTION
**Pull request by ccostin93**

### CODE REVIEW

**Comment by ccostin93** [comment]
I also prepare the changes for the other repos: service, gradle/maven-plugin, but I need to wait for the inspection module to be merged and published first.

File: lib/src/test/java/com/iwillfailyou/inspections/inheritancefree/InheritancefreeTest.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                    "class Temp2 {}",
+                source2
+            )
+        ).intValue();
+        inspection.accept(source1);
+        inspection.accept(source2);
+        final Failures failures = inspection.failures();
+        try {
+            failures.failIfRed();
+            Assert.fail("Must not failed!");
```
Comment (nikialeksey) [inline lib/src/test/java/com/iwillfailyou/inspections/inheritancefree/InheritancefreeTest.java:103]: `"Must failed!"`

File: lib/src/main/java/com/iwillfailyou/inspections/inheritancefree/inheritances/JavaInheritances.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        );
+    }
+
+    public JavaInheritances(
+        final JavaParser parser,
+        final InspectionScalar<InputStream> source,
+        final String descriptor
+    ) {
+        this(
+            new JavaViolations<>(
```
Comment (nikialeksey) [inline lib/src/main/java/com/iwillfailyou/inspections/inheritancefree/inheritances/JavaInheritances.java:105]: Let's implement it other way, see the [`NoMultipleReturn`](https://github.com/iwillfailyou/java-inspections/blob/master/lib/src/main/java/com/iwillfailyou/inspections/nomultiplereturn/NoMultipleReturn.java) and [`MultipleReturnViolations`](https://github.com/iwillfailyou/java-inspections/blob/master/lib/src/main/java/com/iwillfailyou/inspections/nomultiplereturn/multiplereturns/MultipleReturnViolations.java) - it has more flexible approach

**Comment by ccostin93** [comment]
@nikialeksey, why not use `merge squash` strategy so that the master branch commit history becomes cleaner?

**Pull request merged**